src/renderer/utils/numbers.ts

```diff
@@ -27,3 +27,23 @@ export const numberFormatter = (value: any, unit = '', fractionDigits = 2) => {
 
     return `${formatter.format(parsedValue)}${unit}`;
 };
+
+
+export const toHex = (num: number): string => {
+    if (num < 0) {
+        return toHex(0xffffffff + num + 1);
+    }
+    return `0x${num.toString(16).toUpperCase()}`;
+};
```

Comment:
```suggestion
export const toHex = (num: number): string => {
    return `0x${num.toString(16).padStart(8, '0').toUpperCase()}`;
};
```

Add a padding instead, it is simpler to reason about and the method is very well supported

Reply:
num = -1
`0x${num.toString(16).padStart(8, '0').toUpperCase()}` 
`0x000000-1` - this is wrong, no?

Reply:
Use [`Math.sign`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Math/sign) for negative checking and [`Math.abs`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Math/abs) to get the number without the sign. That way you have a more explicit check for negative numbers and can handle the case adding a `-` ins front of the `0x`. I think wrapping the address around will give the wrong number

Reply:
that suggestion woudl be incorrect. you must wrap around the 32 bit unsigned int. negative hex do not exist